pages/Product/components/Content/index.jsx

```diff
@@ -31,93 +38,106 @@ class ProductContent extends Component {
+      variantId: isBaseProduct ? null : this.props.id,
+    });
   }
 
   /**
-   * Sets the variant ID.
-   * @param {string} variantId The selected variant ID.
+   * @param {Object} nextProps The next component props.
+   * @param {Object} nextState The next state.
+   * @return {boolean}
    */
-  handleSetVariantId = (variantId) => {
-    if (variantId !== this.state.variantId) {
-      this.setState({ variantId });
-    }
+  shouldComponentUpdate(nextProps, nextState) {
+    return (
+      this.props.id !== nextProps.id
+      || this.state.productId !== nextState.productId
+      || this.state.variantId !== nextState.variantId
+    );
   }
 
   /**
    * @return {JSX}
    */
   render() {
+    const { productId, variantId } = this.state;
+
+    if (!productId && !variantId) {
+      return null;
+    }
+
     return (
-      <div data-test-id={`product: ${this.props.name}`}>
-        {/* IMAGE */}
-        {/* <Portal name={portals.PRODUCT_IMAGE_BEFORE} />
-        <Portal name={portals.PRODUCT_IMAGE}>
-          <ImageSlider />
-        </Portal>
-        <Portal name={portals.PRODUCT_IMAGE_AFTER} /> */}
-
-        {/* HEADER */}
-        {/* <Portal name={portals.PRODUCT_HEADER_BEFORE} />
-        <Portal name={portals.PRODUCT_HEADER}>
-          <Header />
-        </Portal>
-        <Portal name={portals.PRODUCT_HEADER_AFTER} /> */}
-
-        {/* VARIANT SELECT */}
-        {/* <Portal name={portals.PRODUCT_VARIANT_SELECT_BEFORE} />
-        <Portal name={portals.PRODUCT_VARIANT_SELECT}>
-          <VariantSelects />
-        </Portal>
-        <Portal name={portals.PRODUCT_VARIANT_SELECT_AFTER} /> */}
-
-        {/* OPTIONS */}
-        {/* <Portal name={portals.PRODUCT_OPTIONS_BEFORE} />
-        <Portal name={portals.PRODUCT_OPTIONS}>
-          <Options />
-        </Portal>
-        <Portal name={portals.PRODUCT_OPTIONS_AFTER} /> */}
-
-        {/* DESCRIPTION */}
-        {/* <Portal name={portals.PRODUCT_DESCRIPTION_BEFORE} />
-        <Portal name={portals.PRODUCT_DESCRIPTION}>
-          <Description />
-        </Portal>
-        <Portal name={portals.PRODUCT_DESCRIPTION_AFTER} /> */}
-
-        {/* PROPERTIES */}
-        {/* <Portal name={portals.PRODUCT_PROPERTIES_BEFORE} />
-        <Portal name={portals.PRODUCT_PROPERTIES}>
-          <Properties />
-        </Portal>
-        <Portal name={portals.PRODUCT_PROPERTIES_AFTER} /> */}
-
-        {/* REVIEWS */}
-        {/* <Portal name={portals.PRODUCT_REVIEWS_BEFORE} />
-        <Portal name={portals.PRODUCT_REVIEWS}>
-          <Reviews />
-        </Portal>
-        <Portal name={portals.PRODUCT_REVIEWS_AFTER} /> */}
-
-        {/* TAX DISCLAIMER */}
-        {/* <Portal name={portals.PRODUCT_TAX_DISCLAIMER_BEFORE} />
-        <Portal name={portals.PRODUCT_TAX_DISCLAIMER}>
-          <TaxDisclaimer />
-        </Portal>
-        <Portal name={portals.PRODUCT_TAX_DISCLAIMER_AFTER} /> */}
-      </div>
+      <ProductContext.Provider value={{ productId, variantId }}>
```

Comment:
Instead of creating this inline entity you could simply pass in `this.state` => performance.

Reply:
Passed in full state object